fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/builder/SoundTypeBuilder.java

```diff
@@ -0,0 +1,183 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.builder;
+
+import com.mojang.serialization.Codec;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.include.com.google.common.base.Preconditions;
+
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.StringIdentifiable;
+
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricSoundsProvider;
+import net.fabricmc.fabric.impl.datagen.SoundTypeBuilderImpl;
+
+/**
+ * Provides functionality for building entries that go into a sounds.json file.
+ *
+ * <p>Use in conjunction with {@link FabricSoundsProvider} to generate sound definitions.
+ *
+ * @see net.minecraft.client.sound.SoundManager
+ * @see net.minecraft.client.sound.WeightedSoundSet
+ */
+public interface SoundTypeBuilder {
+	/**
+	 * Creates a new builder pre-filled with a subtitle translation string based on the passed event.
+	 *
+	 * @param event The sound event.
+	 * @return New sound type builder
+	 */
+	static SoundTypeBuilder of(SoundEvent event) {
+		Preconditions.checkArgument(event != null, "Sound event must not be null.");
+		return of().subtitle("subtitles." + event.id().getNamespace() + "." + event.id().getPath());
+	}
+
+	/**
+	 * Creates a new empty builder.
+	 *
+	 * @param event The sound event.
+	 * @return New sound type builder
+	 */
+	static SoundTypeBuilder of() {
+		return new SoundTypeBuilderImpl();
+	}
+
+	/**
+	 * Sets the sound category the sound event must play on.
+	 *
+	 * <p>The default category is {@link SoundCategory#NEUTRAL}. GUI elements should use {@link SoundCategory#MASTER}.
+	 */
+	SoundTypeBuilder category(SoundCategory category);
+
+	/**
+	 * Sets an optional translation string to use for the sound's subtitle.
+	 *
+	 * <p>The default is null (no subtitle).
+	 */
+	SoundTypeBuilder subtitle(@Nullable String subtitle);
+
+	/**
+	 * Adds one sound to the event.
+	 */
+	SoundTypeBuilder sound(EntryBuilder sound);
+
+	/**
+	 * Adds one or more sounds to the event.
+	 * This is a shorthand method for quickly adding multiple entries where
+	 * each sound is a variant with an index.
+	 *
+	 * @param sound The base sound to add.
+	 * @param count The number of instances of that sound to register.
+	 */
+	SoundTypeBuilder sound(EntryBuilder sound, int count);
+
+	/**
+	 * @see net.minecraft.client.sound.Sound.RegistrationType
```

Comment:
Remove; use vanilla

Reply:
Vanilla doesn't implement StringConvertible. I needed a codec.

Reply:
Looking at it again, I can do what I did for SoundCategory (that also does not have a codec for some reason) but I'd like for some javadocs to be added to the Sound.RegistrationType enum as it's currently undocumented.